imgui/src/widget/list_box.rs

```diff
@@ -4,42 +4,22 @@ use crate::string::ImStr;
 use crate::sys;
 use crate::Ui;
 
-#[derive(Copy, Clone, Debug)]
-enum Size {
-    Vec(sys::ImVec2),
-    Items {
-        items_count: i32,
-        height_in_items: i32,
-    },
-}
 /// Builder for a list box widget
 #[derive(Copy, Clone, Debug)]
 #[must_use]
 pub struct ListBox<'a> {
     label: &'a ImStr,
-    size: Size,
+    size: sys::ImVec2,
 }
 
 impl<'a> ListBox<'a> {
     /// Constructs a new list box builder.
     pub const fn new(label: &'a ImStr) -> ListBox<'a> {
         ListBox {
             label,
-            size: Size::Vec(sys::ImVec2::zero()),
+            size: sys::ImVec2::zero(),
         }
     }
-    /// Sets the list box size based on the number of items that you want to make visible
-    /// Size default to hold ~7.25 items.
-    /// We add +25% worth of item height to allow the user to see at a glance if there are more items up/down, without looking at the scrollbar.
-    /// We don't add this extra bit if items_count <= height_in_items. It is slightly dodgy, because it means a dynamic list of items will make the widget resize occasionally when it crosses that size.
-    #[inline]
-    pub const fn calculate_size(mut self, items_count: i32, height_in_items: i32) -> Self {
```

Comment:
Hmm, if it's deprecated in imgui 1.81, maybe we should just deprecate it and remove in a future version? Unless there's a compelling reason to just remove it outright

Reply:
Main reason I removed it was the binding function disappeared, and I didn't put much thought into why..

Looking into it now, cimgui defines `IMGUI_DISABLE_OBSOLETE_FUNCTIONS` by default so you don't get obsolete functions.

Thoughts?

Reply:
Fair enough. We'll need to remember to mention it in the changelog, then.